jni/src/main/java/com/horizen/storage/Storage.java

```diff
@@ -0,0 +1,114 @@
+package com.horizen.storage;
+
+import com.horizen.common.ColumnFamily;
+import com.horizen.librust.Library;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class Storage implements AutoCloseable {
+    // Loading the Rust library which contains all the underlying logic
+    static {
+        Library.load();
+    }
+
+    private long storagePointer;
+
+    public void checkPointer() throws IllegalStateException {
+        if (storagePointer == 0)
+            throw new IllegalStateException("Storage instance was freed");
+    }
+
+    // Constructor is intended to be called from inside of the Rust environment for setting a raw pointer to a Rust-instance of Storage
+    private Storage(long storagePointer) {
+        this.storagePointer = storagePointer;
+    }
+
+    // Gates to the Rust-side API
+    private static native Storage nativeOpen(String storagePath, boolean createIfMissing);
+    private static native void nativeClose(long storagePointer);
+
+    private native byte[] nativeGet(ColumnFamily cf, byte[] key);
+    private native Map<byte[], Optional<byte[]>> nativeMultiGet(ColumnFamily cf, byte[][] keys);
+    private native boolean nativeIsEmpty(ColumnFamily cf);
```

Comment:
let's add also `isEmpty()` without any CF argument. Corresponing rust method is `fn is_empty(&self) -> bool {
        self.get_iter().next().is_none()
    }`
Same for Transaction

Reply:
Still actual. Let's add it.

Reply:
Done